examples/emberjs/todomvc/app/components/todo-item.gjs

```diff
@@ -0,0 +1,68 @@
+import Component from '@glimmer/component';
+import { on } from '@ember/modifier';
+import { service } from '@ember/service';
+import { isBlank } from '@ember/utils';
+
+export default class TodoItem extends Component {
+  <template>
+    <li class="{{if @todo.completed 'completed'}} {{if this.editing 'editing'}}">
+      <div class="view">
+        <input
+          class="toggle"
+          type="checkbox"
+          aria-label="Toggle the completion state of this todo"
+          checked={{@todo.completed}}
+          {{on 'change' this.toggleCompleted}}
+        >
+        <label {{on 'dblclick' this.startEditing}}>{{@todo.title}}</label>
+        <button class="destroy" {{on 'click' this.removeTodo}} type="button"></button>
+      </div>
+      <input
+        class="edit"
+        value={{@todo.title}}
+        {{on 'blur' this.doneEditing}}
+        {{on 'keydown' this.handleKeydown}}
+        autofocus
+      >
+    </li>
+  </template>
+
+  @service repo;
+
+  removeTodo = () => this.repo.delete(this.args.todo);
+
+  toggleCompleted = (event) => {
+      this.args.todo.completed = event.target.checked;
+			this.repo.persist();
+  }
+
+  handleKeydown = (event) => {
+    if (event.keyCode === 13) {
+      event.target.blur();
+    } else if (e.keyCode === 27) {
+      this.editing = false;
+    }
+  }
+
+		startEditing = (event) => {
+			this.args.onStartEdit();
+      this.editing = true;
+
+      event.closest('li')?.querySelector('input.edit').focus();
```

Comment:
I think you want `event.target.closest` here, right?

Reply:
doh, yup. thanks!